misaligned.spec.js

```diff
@@ -0,0 +1,45 @@
+const {expect} = require('chai');
+const  print_color_map = require('./misaligned.js');
+// Import the print_color_map function
+// Redirect console.log to a custom function
+const originalLog = console.log;
+const capturedLogs = [];
+
+console.log = function (...args) {
+    capturedLogs.push(args.join(' '));
+    originalLog.apply(console, args);
+};
+
+//console.log(capturedLogs);
+
+ describe('print_color_map Function', () => {
+    it('should have aligned "|" characters', () => {
+        
+        // Call the function to print the color map
+        result = print_color_map();
+
+        // Get the captured console output
+        const output = capturedLogs;
+
+        // Check if the "|" characters are aligned in each line
+        for (let k = 0; k < output.length - 1; k++) {
+            const currentLine = output[k];
+            const nextLine = output[k + 1];
+            const pipeIndex = currentLine.indexOf('|');
+            const nextPipeIndex = nextLine.indexOf('|');
+
+            expect(pipeIndex).to.equal(nextPipeIndex);
+        }
+    });
+
+    it('should return the total number of color combinations', () => {
+        expect(result).to.equal(25);
```

Comment:
This test depends on the result set in the previous test. Unit tests need to be independent of each other.
Also, you have a more powerful way of checking the number of rows in the console, which is better than the 'indirect' check of the return value of the printing function

Reply:
Thanks for pointing it out. Will experiment with this more powerful method :)